### DISCUSSION TITLE
Pybricks in VS code

### DISCUSSION THREAD

**Discussion by SaharshSS** [pybricks/support#1299] [Q&A] [answered]
Hi,
I am unable to get pybricks to work in vs code. I followed all the steps described online, but whenever I run pybricksdev it just says that pybricks is undefined. Does anyone know how to get it to work properly?

**Comment by FLL-Team-24277** [comment, marked as answer]
In our Repo here:
https://github.com/FLL-Team-24277/Master-Program-Fall-2023
there's a folder called "help". In there is a PowerPoint file called "Laptop Setup". It goes through the entire setup that we used. The repo and help files are geared toward FIRST Lego League teams, but it's pretty easy to ignore the stuff that isn't applicable.

But yes, we use pybricks ENTIRELY on VS Code. We have never used the online version of pybricks.

Specifically, for your problem it sounds to me like you may have a virtual environment set up for your project, and maybe you haven't pip installed pybricks and pybricksdev?

**Reply by SaharshSS** [reply]
I did pip install pybricks and pybricksdev, but I'll check your repo. Thanks for the response!

**Comment by dlech** [comment]
> I followed all the steps described online

Where? 

Do you have the Python extension installed?

If you are using the LEGO extension for EV3 to generate a project template, did you also enable the Python extension in `.vscode/settings.json` in your project folder?

**Comment by ksjzang** [comment]
i have same question! I want to write to spike hub. Is there exist solution?

**Comment by AnandSingh** [comment]
I created an Pybricks Runner, a VSCode extension that lets you effortlessly program LEGO SPIKE Prime, EV3, and other Pybricks-compatible bricks directly within Visual Studio Code!

Check out the extension and installation guide here:
https://github.com/AnandSingh/pybricks-runner-vscode

Your feedback and contributions are warmly welcome! Feel free to open issues or submit pull requests.
